lib/src/dag_walk.rs

```diff
@@ -43,10 +43,10 @@ where
 }
 
 /// Returns neighbors before the node itself.
-pub fn topo_order_reverse<'a, T, ID, II, NI>(
+pub fn topo_order_reverse<T, ID, II, NI>(
```

Comment:
nit: It's probably clearer to also replace `II`, `ID`, and `NI` by `impl ...`. That's mostly unrelated to this PR, though, and can be done separately.

Reply:
Some of these type parameters appear in `Fn`/`FnMut` trait bound, so we can't remove all of them.